packages/shared/src/components/modals/CompanionModal.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { ReactElement, useMemo } from 'react';
+
+import { LazyModalCommonProps, Modal } from './common/Modal';
+import { Button } from '../buttons/Button';
+import UpvoteIcon from '../icons/Upvote';
+import CommentIcon from '../icons/Discuss';
+import LineIcon from '../icons/Line';
+import { IconSize } from '../Icon';
+import useMedia from '../../hooks/useMedia';
+import { mobileL, tablet } from '../../styles/media';
+import { cloudinary } from '../../lib/image';
+
+type Props = LazyModalCommonProps & {
+  url: string;
+  onReadArticleClick: (any) => void;
+  onActivateCompanion: () => void;
+};
+
+export default function CompanionModal({
+  onReadArticleClick,
+  onActivateCompanion,
+  url,
+  ...props
+}: Props): ReactElement {
+  const isTablet = !useMedia([tablet.replace('@media ', '')], [true], false);
+  const isMobile = !useMedia([mobileL.replace('@media ', '')], [true], false);
+
+  const size = useMemo(() => {
+    if (isMobile) return Modal.Size.XSmall;
+    if (isTablet) return Modal.Size.Small;
+
+    return Modal.Size.XLarge;
+  }, [isTablet, isMobile]);
+
+  return (
+    <Modal
+      {...props}
+      className="border-none"
+      kind={Modal.Kind.FlexibleCenter}
+      size={size}
+    >
+      {isTablet && (
+        // eslint-disable-next-line jsx-a11y/media-has-caption
+        <video
+          className="tablet:hidden rounded-16 w-fit"
+          poster={cloudinary.companion.videoThumbnailMobile}
```

Comment:
I would like to rather see the poster and src change based on the isTablet, think it will cleanup the duplicate video a bit.

Reply:
I actually did try that, but it turns out that if you update the `src` on a video while it's in a loop, it actually keeps playing the initial video instead of switching to the new one. 

For this reason, I kept it as is, even though not ideal. I'm open to other suggestions though